modules/library/render/src/test/java/org/geotools/renderer/lite/MemoryFilterOptimizerTest.java

```diff
@@ -144,4 +148,35 @@ public void testEqualFeatureTypes() throws Exception {
         Mockito.verify(spy, Mockito.times(0)).getAttribute(name);
         Mockito.verify(spy, Mockito.times(1)).getAttribute(0);
     }
+
+    public void testInFunctionOptimizer() throws Exception {
```

Comment:
This test covers only the happy case, could you write tests for the exception paths, like or with different properties, and with comparisons other than equality?

Reply:
Agreed, added test cases when the In function optimization is not accomplished.